exist-core/src/main/java/org/exist/xquery/Function.java

```diff
@@ -56,98 +53,67 @@
 public abstract class Function extends PathExpr {
 
     // Declare it in Namespaces instead? /ljo
-    public final static String BUILTIN_FUNCTION_NS =
-            "http://www.w3.org/2005/xpath-functions";
+    public final static String BUILTIN_FUNCTION_NS = "http://www.w3.org/2005/xpath-functions";
 
     /**
      * The module that declared the function.
      */
     protected Module parentModule;
-
+    /**
+     * Flag to indicate if argument types are statically checked.
+     * This is set to true by default (meaning: no further checks needed).
+     * Method {@link #setArguments(java.util.List)} will set it to false
+     * (unless overwritten), thus enforcing a check.
+     */
+    protected boolean argumentsChecked = true;
```

Comment:
From reading the JavaDoc and the naming of the variable. It is not clear to me whether `argumentsChecked` means that: "The arguments have already been checked", or whether it means that "The arguments will need to be checked". Perhaps some rewording/renaming here would help?

Reply:
Also, is `protected` scope correct here?